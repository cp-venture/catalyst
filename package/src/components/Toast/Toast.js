import React from "react";
import PropTypes from "prop-types";
import { Snackbar } from "@material-ui/core";
import { ToastWrapper } from "./helpers";

/**
 * @name Toast
 * @param {Object} props Component props
 * @returns {React.Component} returns a React component
 */
const Toast = React.forwardRef(function Toast(props, ref) {
  const { message, variant, title, ...otherProps } = props;
  const [open, setOpen] = React.useState(false);

  const handleClose = (event, reason) => {
    if (reason === "clickaway") {
      return;
    }
    setOpen(false);
  };

  return (
    <Snackbar
      ref={ref}
      open={open}
      {...otherProps}
    >
      <ToastWrapper
        props={otherProps}
        variant={variant}
        title={title}
        message={message}
        onClose={handleClose}
      />
    </Snackbar>
  );
});


Toast.propTypes = {
  /**
   * Message: Node
   */
  message: PropTypes.node,
  /**
   * Title: Optional
   */
  title: PropTypes.string,
  /**
   * Variant: Info, Success, Warning, Error
   */
  variant: PropTypes.oneOf(["error", "info", "success", "warning"]).isRequired
};

Toast.defaultProps = {
  variant: "info"
};

export default Toast;
