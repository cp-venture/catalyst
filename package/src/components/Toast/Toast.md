### Overview

<!-- Get this short paragraph from design. -->

The X component inherits from the Material-UI [X component](https://material-ui.com/components/X/). Refer to the Material-UI [X docs](https://material-ui.com/api/X/) for more information.

### Usage

<!-- Show all the variants/combos we use in Reaction Admin, without the code box > -->

```jsx noeditor
import Button from "../Button";
import IconButton from "@material-ui/core/IconButton";
import CloseIcon from "mdi-material-ui/Close";

function OpenToast() {
  const [open, setOpen] = React.useState(false);

  const handleClick = () => {
    setOpen(true);
  };

  const handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }

    setOpen(false);
  };

  return (
    <div>
      <Button variant="contained" color="primary" onClick={handleClick}>Open toast</Button>
      <Toast
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'left',
        }}
        open={open}
        autoHideDuration={6000}
        onClose={handleClose}
        ContentProps={{
          'aria-describedby': 'message-id',
        }}
        message={<span id="message-id">Note archived</span>}
      />
    </div>
  );
}

<OpenToast/>
```

#### Types

<!-- Show all Types of the component used in Reaction Admin -->

##### Name of type

<!-- Explain when to use this type of the component, and give a real life Reaction Admin example. If needed, add instruction for developers on how to set up the component. -->

Use a X component to allow a user to XX, such as XYXY.

```jsx
<Toast/>
```