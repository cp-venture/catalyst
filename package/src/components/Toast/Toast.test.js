import React from "react";
import { render } from "../../tests/index.js";
import Toast from "./Toast";

test("basic snapshot - only default props", () => {
  const { asFragment } = render(<Toast message="Test message" open/>);
  expect(asFragment()).toMatchSnapshot();
});
