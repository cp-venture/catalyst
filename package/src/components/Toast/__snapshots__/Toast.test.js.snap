// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot - only default props 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
  >
    <div
      class="MuiTypography-root MuiPaper-root MuiPaper-elevation6 MuiSnackbarContent-root makeStyles-info-10 MuiTypography-body2"
      direction="up"
      props="[object Object]"
      role="alertdialog"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiSnackbarContent-message"
      >
        Test message
      </div>
    </div>
  </div>
</DocumentFragment>
`;
